src/core/status.h

```diff
@@ -51,6 +52,9 @@ class Status {
   // Construct a status from a code and message.
```

Comment:
Do we need to replicate the Code class above. Can't you just inherit that from common::Error? Can't everything be inherited except the TRITONSERVER_* conversions?

Reply:
Do we want to move `TRITONSERVER_*` conversions to `common::Error` as well? For users of `common::Error`, i.e. implementing a backend, it is likely that they will need to convert it to `TRITONSERVER_Error` at the end

Reply:
@GuanLuo I was thinking in that direction. Like moving status.cc/.h to common-repo from core. But it will expose the entire TRTITONSERVER_*  api to all the places where status.h is being used. I decided against this for better encapsulation. It would be good to have self-sufficient utilities in common-repo. 

However, I do see benefit in breaking the encapsulation and including TRITONSERVER_* conversions in `triton::common::`. If you find it a better solution, I am in for migrating entire `status.cc/status.h` to common::repo.